test/martha/martha_v3.test.js

```diff
@@ -279,30 +536,86 @@ test.serial('martha_v3 calls bond Bond with the "fence" provider when the Data O
 });
 
 test.serial('martha_v3 does not call Bond or return SA key when the Data Object URL host endswith ".humancellatlas.org', async (t) => {
-    getJsonFromApiStub.onFirstCall().resolves(dataObjectServiceObject);
+    getJsonFromApiStub.onFirstCall().resolves(jadeDosResponse);
     const response = mockResponse();
     await marthaV3(mockRequest({ body: { 'url': 'drs://someservice.humancellatlas.org/this_part_can_be_anything' } }), response);
     const result = response.send.lastCall.args[0];
     t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
-    t.deepEqual(Object.assign({}, result), fullExpectedResult(null));
+    t.deepEqual(Object.assign({}, result), jadeDosMarthaResult(null));
     t.falsy(result.googleServiceAccount);
     t.is(response.statusCode, 200);
 });
 
 test.serial('martha_v3 does not call Bond or return SA key when the host url is for jade data repo', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(jadeDrsResponse);
+    const response = mockResponse();
+    await marthaV3(mockRequest({ body: { 'url': 'drs://jade.datarepo-dev.broadinstitute.org/abc' } }), response);
+    const result = response.send.lastCall.args[0];
+    t.true(getJsonFromApiStub.calledOnce); // Bond was not called to get SA key
+    t.deepEqual(Object.assign({}, result), jadeDosMarthaResult(null));
+    t.falsy(result.googleServiceAccount);
+    t.is(response.statusCode, 200);
+});
+
+test.serial('martha_v3 parses Gen3 CRDC response correctly', async (t) => {
+    getJsonFromApiStub.onFirstCall().resolves(gen3CrdcResponse);
```

Comment:
As Saloni said, but for clarity, this test should call gen3, not JDR. I'm commenting on this one specifically since the line of code is outside of the diff.